pipelines/ApPipeWithFakes.yaml

```diff
@@ -31,45 +25,109 @@ tasks:
     class: lsst.pipe.tasks.insertFakes.InsertFakesTask
     config:
       connections.coaddName: parameters.coaddName
+      connections.fakesType: parameters.fakesType
       doSubSelectSources: True
       select_col: 'isTemplateSource'
   processVisitFakes:
     class: lsst.pipe.tasks.processCcdWithFakes.ProcessCcdWithVariableFakesTask
     config:
       connections.coaddName: parameters.coaddName
+      connections.fakesType: parameters.fakesType
       insertFakes.doSubSelectSources: True
       insertFakes.select_col: 'isVisitSource'
       calibrate.photoCal.match.referenceSelection.magLimit.fluxField: i_flux
       calibrate.photoCal.match.referenceSelection.magLimit.maximum: 22.0
-  imageDifference:
+  retrieveTemplateWithFakes:
+    class: lsst.ip.diffim.getTemplate.GetTemplateTask
+    config:
+      connections.coaddName: parameters.coaddName
+      connections.fakesType: parameters.fakesType
+  imageDifferenceWithFakes:
     class: lsst.pipe.tasks.imageDifference.ImageDifferenceFromTemplateTask
     config:
+      connections.coaddName: parameters.coaddName
+      coaddName: parameters.coaddName  # Can be removed once ImageDifference no longer supports Gen 2
       connections.fakesType: parameters.fakesType
-      connections.exposure: fakes_calexp
-  transformDiaSrcCat:
+      doWriteWarpedExp: True             # Required for packaging alerts in diaPipe
+      doSkySources: True
+  transformDiaSrcCatWithFakes:
     class: lsst.ap.association.TransformDiaSourceCatalogTask
     config:
+      connections.coaddName: parameters.coaddName
       connections.fakesType: parameters.fakesType
+      doRemoveSkySources: True
   diaPipe:
     class: lsst.ap.association.DiaPipelineTask
     config:
       doWriteAssociatedSources: True
+      connections.coaddName: parameters.coaddName
       connections.fakesType: parameters.fakesType
       # apdb.db_url: YOU MUST CONFIGURE THIS BEFORE RUNNING THE PIPELINE
   fakesMatch:
-    class: lsst.pipe.tasks.matchFakes.MatchFakesTask
+    class: lsst.pipe.tasks.matchFakes.MatchVariableFakesTask
     config:
       connections.coaddName: parameters.coaddName
+      connections.fakesType: parameters.fakesType
       matchDistanceArcseconds: 0.5
+subsets:
+  apPipe:
```

Comment:
I think there needs to be a `processCcd` subset in here somewhere, because the camera-specific pipelines exclude it (in favor of camera-specific `processCcd`). Or maybe that is handled when `ProcessCcd.yaml` is imported up at the top? Mostly I am confused because `apPipe` as a subset looks like what my brain categorizes as just `processCcd`.

Basically, please convince me that the camera-specific `ApPipeWithFakes` pipelines are handling where they import "processCcd" steps and from correctly.

Reply:
I think `processCcd` is unchanged from the import, since we don't exclude anything from `ProcessCcd.yaml`. I could add a comment here to that effect.

I'm not sure what to do about the `apPipe` subset. I originally wrote it intending to add a non-fakes `imageDifference` step, but later decided that would be silly (the only reason we need such a task is so that certain metrics won't be contaminated by fakes). I want the subsets to be consistent with what `ap_verify` is using, so maybe just `prepareFakes` and `apPipeWithFakes`?

Reply:
I'm not sure what to do about the `apPipe` subset. I originally wrote it intending to add a non-fakes `imageDifference` step to `ApPipeWithFakes`, but later decided that would be silly (the only reason we need such a task is so that certain `ap_verify` metrics won't be contaminated by fakes). I want the subsets to be consistent with what `ApVerifyWithFakes` is using, so maybe just `prepareFakes` and `apPipeWithFakes`?

Reply:
I honestly never use the subsets, which is probably why there weren't any in here to begin with. In my mind, though, an `apPipe` subset ought to be defined as processCcd + imageDifference + diaPipe (where processCcd = isr + characterize + calibrate). Defining it as such - and calling it whatever you like, e.g., either `apPipe` or `apPipeWithFakes` - would work fine for me.

Reply:
And sure, a `prepareFakes` subset sounds useful!

Reply:
Ok, I'll remove the `apPipe` subset then.